chore: Add matamo integration doc

farhaanbukhsh opened the pull request:
## Description

This PR aims to help add documentation regarding adding analytics to the platform apart from Google Analytics.

Useful information to include:


## Supporting information


## Testing instructions

Nothing to test

## Deadline

"None"

## Other information

Include anything else that will help reviewers and consumers understand the change.



Private ref: [BB-9388](https://tasks.opencraft.com/browse/BB-9388)

openedx-webhooks commented:
Thanks for the pull request, @farhaanbukhsh!

### What's next?

*Please work through the following steps to get your changes ready for engineering review:*

#### :radio_button: Get product approval

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.

#### :radio_button: Provide context

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads





#### :radio_button: Get a green build

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.



#### :radio_button: Let us know that your PR is ready for review:

### Who will review my changes?


This repository is currently maintained by `@openedx/docs-openedx-org-maintainers`.  Tag them in a comment and let them know that your changes are ready for review.


### Where can I find more information?

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)

### When can I expect my changes to be merged?

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*

<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

sarina commented on source/site_ops/how-tos/matamo-analytics.rst:
```diff
@@ -0,0 +1,20 @@
+2. Add the script provided by the platform to the file.
+3. Apply the theme, and we are good to go.
+
+Note::
```
```suggestion
.. note::
```

and the following line will need indentation. See https://docs.openedx.org/en/latest/documentors/references/quick_reference_rst.html#directives

xitij2000 reviewed (approved):
Generally looks good, but I would recommend a few changes. 

1. Don't focus this on Matamo, but use Matamo as an example. What I mean is, make this a generic document about how to integrate any analytics system with Open edX and then give Matamo as one such example. 
2. Give some actual example code. i.e. with a fake Matamo domain etc. so it's clearer what needs to be done.

farhaanbukhsh commented:
> Generally looks good, but I would recommend a few changes.
> 
>     1. Don't focus this on Matamo, but use Matamo as an example. What I mean is, make this a generic document about how to integrate any analytics system with Open edX and then give Matamo as one such example.
> 
>     2. Give some actual example code. i.e. with a fake Matamo domain etc. so it's clearer what needs to be done.

Addressed those comments @xitij2000 , Thank you for the review.

sarina commented on source/site_ops/how-tos/matamo-analytics.rst:
```diff
@@ -0,0 +1,49 @@
+Configuring Analytics
```
I think this title is misleading. When I saw it in the TOC, I thought it was about enabling Aspects and was confused why I couldn't find the doc on Matamo. The file name and the document title should match. So this should be called "Matamo Analytics", or it could be called "Configuring Site-wide Analytics Tools" and the doc title made to match.

farhaanbukhsh replied:
I agree @sarina I was also a bit on the fence about the title so thanks a lot for your suggestion I have named is a site-wide analytics tool. This makes more sense.

sarina commented on source/site_ops/how-tos/matamo-analytics.rst:
```diff
@@ -0,0 +1,49 @@
+
+Matamo needs to be included under the head tag of the page; hence, we will use head-extra.html to include the analytics script.
+
+1. Create the head-extra.html file under {theme}/lms/templates/.
```
Use `#.`

farhaanbukhsh replied:
done

sarina commented on source/site_ops/how-tos/matamo-analytics.rst:
```diff
@@ -0,0 +1,49 @@
+1. Create the head-extra.html file under {theme}/lms/templates/.
+2. Add the script provided by the platform to the file.
+
+For example:
```
Indent this example at the same level as the numbered bullet.

farhaanbukhsh replied:
done

xitij2000 commented:
@sarina  Thanks! This looks good to me as well.
